zcash_primitives/src/transaction/components/sapling/builder.rs

```diff
@@ -491,7 +486,18 @@ impl<P: consensus::Parameters> SaplingBuilder<P> {
             })
             .collect();
 
-        Ok((spend_descs, output_descs, tx_metadata))
+        let bundle = if shielded_spends.is_empty() && shielded_outputs.is_empty() {
+            None
+        } else {
+            Some(Bundle {
+                shielded_spends,
+                shielded_outputs,
+                value_balance: self.value_balance,
+                binding_sig: (),
+            })
```

Comment:
```suggestion
            Some((Bundle {
                shielded_spends,
                shielded_outputs,
                value_balance: self.value_balance,
                binding_sig: (),
            }, tx_metadata))
```

Reply:
I'm going to defer this until we move signature application into the bundles themselves.